cluster-autoscaler/simulator/clustersnapshot/basic.go

```diff
@@ -234,49 +225,47 @@ func (snapshot *BasicClusterSnapshot) AddNodeInfo(nodeInfo *framework.NodeInfo)
 		return err
 	}
 	for _, podInfo := range nodeInfo.Pods() {
-		if err := snapshot.getInternalData().addPod(podInfo.Pod, nodeInfo.Node().Name); err != nil {
+		if err := snapshot.getInternalData().schedulePod(podInfo.Pod, nodeInfo.Node().Name); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// AddNode adds node to the snapshot.
-func (snapshot *BasicClusterSnapshot) AddNode(node *apiv1.Node) error {
-	return snapshot.getInternalData().addNode(node)
-}
-
-// AddNodes adds nodes in batch to the snapshot.
-func (snapshot *BasicClusterSnapshot) AddNodes(nodes []*apiv1.Node) error {
-	return snapshot.getInternalData().addNodes(nodes)
-}
+// Initialize initializes the snapshot.
+func (snapshot *BasicClusterSnapshot) Initialize(nodes []*apiv1.Node, scheduledPods []*apiv1.Pod) error {
+	snapshot.Clear()
```

Comment:
Shouldn't we just remove Clear() from interface at this point? It feels completely redundant. Initialize(nil, nil) should do exactly the same thing and, after renaming it, should also be a very obvious and natural way of doing that (not to mention the fact that IIRC we never Clear() a snapshot without immediately re-initializing it, so the Clear() functionality is probably never needed).